controllers/users.js

```diff
@@ -34,9 +34,33 @@ function update(req, res, next) {
     .catch((err) => next(err))
 }
 
+function setProfile(req, res, next) {
+  const {
+    name, email, password, geolocation,
+  } = req.body
+
+  User.findByIdAndUpdate(req.user.id, {
+    set: {
+      name, email, password, geolocation,
+    },
+  })
+    .exec()
+    .then((user) => res.json(user))
+    .catch((err) => next(err))
+}
+
+function getProfile(req, res, next) {
+  User.findById(req.user.id)
```

Comment:
We can simply response req.user, because we've findById in authenticate middleware. If there are any errors, they will be thrown after go here.